contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -5278,3 +5289,85 @@ is_partitioned_table_reloptions_allowed(Datum reloptions)
 	}
 	return true;
 }
+
+static bool
+is_babelfish_builtin_type(Oid typid)
+{
+	bool res = false;
+	HeapTuple	tp;
+	tp = SearchSysCache1(TYPEOID, ObjectIdGetDatum(typid));
+	if (HeapTupleIsValid(tp))
+	{
+		Form_pg_type typtup = (Form_pg_type) GETSTRUCT(tp);
+		res = pg_strcasecmp(get_namespace_name(typtup->typnamespace), "sys") == 0;
```

Comment:
We have cached sys schema OID available. Can we use it.
`extern Oid	sys_schema_oid;`

Reply:
The idea here is to check whether the type is in sys schema or not. What you are saying is applicable for bbf builtin datatype but datatypes like text, ntext ...their schema is not sys. Hence this function should return false. If we simply check whether the schema is sys or not using the cached sys_schema_oid, it will always return true which is not expected.

Reply:
My original comment was ambiguous.
Actually I mean skipping get_namespace_name() and reducing one syscache lookup.
so new condition would be
```
res = (typtup->typnamespace == cached_sys_schema_oid);
```